dprital/sonic-buildimage
# Support pulling sonic-slave-docker image from path at REGISTRY_SERVER

#### Why I did it

In order to reduce sonic build time, there is an option to acquire sonic slave docker(s) from artifact server (reduce sonic make configure time).
Current implementation supports only convention of: 

<REGISTRY_SERVER>:<REGISTRY_PORT>/<SLAVE_BASE_IMAGE>:<SLAVE_BASE_TAG>

In case the SLAVE_BASE_IMAGE appear in internal path inside the server, the convention should be like that:

<REGISTRY_SERVER>:<REGISTRY_PORT><REGISTRY_SERVER_PATH>/<SLAVE_BASE_IMAGE>:<SLAVE_BASE_TAG>

When REGISTRY_SERVER_PATH (that is set on rules/config) will have to start with "/".

If REGISTRY_SERVER_PATH will not be set, the behavior will remain the same it works today.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Add ability to set REGISTRY_SERVER_PATH  and update the code for docker image tag and docker image pull accordingly 

#### How to verify it

Use sonic slave docker image from artifact server in which the image is kept in internal folder and make sure it consume it.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
